src/data/set/basic.lean

```diff
@@ -694,6 +723,10 @@ by simp [insert_eq, union_diff_self, -union_singleton, -singleton_union]
 
 @[simp] lemma diff_self {s : set α} : s \ s = ∅ := ext $ by simp
 
+lemma mem_diff_singleton_empty {s : set α} {t : set (set α)} :
```

Comment:
Do we really need this? This is essentially just a combination of `mem_diff` and `mem_singleton_iff`.

Reply:
I don't know. I definitely wanted it for my specific use-case, since I needed it multiple (10?) times. But maybe mathlib is not the place for "combined" facts like this. I will remove it.

Reply:
If you needed it 10 times, I think it's legit to keep it. In my eyes >80% of formalisation libraries consist of little lemmas that are "essentially just a combination"... (not being sarcastic).

Reply:
I would suggest generalizing to `s \in t \ {a} <-> s \in t /\ s != a`, assuming the situation with ne_zero vs nonempty is resolved.